### PR TITLE
Feature/pg error view

### PR DESCRIPTION
**Pull request by tleekkul**

### CODE REVIEW

File: tests/test-pg-error-view.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+    "webserver": {
+        "views/test-pg-error": {
+            "view": "fostgres.control.pg-error",
+            "configuration": {
+                "execute": {
+                    "view": "fostgres.sql",
+                    "configuration": {
+                        "dbname": [
```
Comment (KayEss) [inline tests/test-pg-error-view.json]: These are no longer needed as what you have I think is the default anyway.

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,4 +4,4 @@ project(fostgres)
 add_subdirectory(Cpp)
 add_subdirectory(Example)
 add_subdirectory(Extras)
-
+add_subdirectory(tests)
```
Comment (KayEss) [inline CMakeLists.txt]: Why not add these tests to the existing `Example/users` folder?

File: Cpp/fostgres/fostgres-control-error.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*
+    Copyright 2016, Felspar Co Ltd. http://support.felspar.com/
```
Comment (KayEss) [inline Cpp/fostgres/fostgres-control-error.cpp]: 2019

File: tests/test-pg-error-view.json (hunk 1 above)
Comment (tleekkul) [reply tests/test-pg-error-view.json]: Removed

File: tests/test-pg-error-view.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                                            },
+                                            "friend_type": {}
+                                        }
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                },
+                "": "fost.response.501"
```
Comment (KayEss) [inline tests/test-pg-error-view.json]: I would use a different error code here because the `not_implemented` exception will also give a 501 even in tests.

**Pull request closed without merging**